references/data-api/README.md

```diff
@@ -0,0 +1,150 @@
+### Template the proxy
+
+```sh
+export BQ_PROJECT_ID='my-bq-project'
+export BASE_PATH='/london/bikes/v1'
+export DATA_SET='bigquery-public-data.london_bicycles.cycle_hire'
+
+export PROXY_NAME="$( tr '/' '-' <<< ${BASE_PATH:1})"
+
+rm -rf "./$PROXY_NAME/"
+mkdir "./$PROXY_NAME/"
+cp -r template/ "./$PROXY_NAME/"
+
+find ./$PROXY_NAME -name '*.xml' -print0 |
+while IFS= read -r -d '' file; do
+    echo "replacing variables in $file"
+    envsubst < "$file" > "$file.out"
+    mv "$file.out" "$file"
+done
+
+mv "./$PROXY_NAME/apiproxy/proxy.xml" "./$PROXY_NAME/apiproxy/$PROXY_NAME.xml"
+```
+
+### Deploy the proxy
+
+```sh
+APIGEE_TOKEN=$(gcloud auth print-access-token);
+APIGEE_X_ORG="$PROJECT_ID"
+APIGEE_X_ENV="test1" # set this to the env you want to deploy to
+sackmesser deploy -o $APIGEE_X_ORG -e $APIGEE_X_ENV -d "./$PROXY_NAME" -t "$APIGEE_TOKEN" --deployment-sa "$SA_EMAIL"
+```
+
+### Try it out
+
+```sh
+APIGEE_X_HOSTNAME='my-proxy-hostname'
+```
+
+Basic API request:
+
+```sh
+curl "https://$APIGEE_X_HOSTNAME/london/bikes/v1" -v
+```
+
+Using query params:
+
+```sh
+curl "https://$APIGEE_X_HOSTNAME/london/bikes/v1?limit=3&fields=start_station_name,end_station_name" | jq
+```
+
+Triggering the quota limit for unauthenticated users:
+
+```sh
+curl "https://$APIGEE_X_HOSTNAME/london/bikes/v1?try=[1-10]" -I
+```
+
+Create an API product with a higher quota and an app for it.
+
+```sh
+# Create a Developer Resource
+curl -X POST "https://apigee.googleapis.com/v1/organizations/$APIGEE_X_ORG/developers" \
+-H "Authorization: Bearer $APIGEE_TOKEN" \
+-H "Content-Type: application/json" \
+-d '{ "email": "testdatauser@example.com", "firstName": "test", "lastName": "user", "userName": "testdatauser" }'
+
+# Create an API Product with a Quota higher than the default 8 requests per minute
+curl -X POST "https://apigee.googleapis.com/v1/organizations/$APIGEE_X_ORG/apiproducts" \
+-H "Authorization: Bearer $APIGEE_TOKEN" \
+-H "Content-Type: application/json" \
+--data @<(cat <<EOF
+{
+  "name": "premium-data",
+  "quota": "30",
+  "quotaTimeUnit": "minute",
+  "quotaInterval": "1",
+  "proxies": [
+    "$PROXY_NAME"
+  ],
+  "environments": [
+    "$APIGEE_X_ENV"
+  ],
+  "displayName": "Premium Data Product",
+  "approvalType": "auto"
+}
+EOF
+)
+
+# Create an App
+curl -X POST "https://apigee.googleapis.com/v1/organizations/$APIGEE_X_ORG/developers/testdatauser@example.com/apps" \
+-H "Authorization: Bearer $APIGEE_TOKEN" \
+-H "Content-Type: application/json" \
+--data @<(cat <<EOF
+{
+  "name": "premium-data-app",
+  "apiProducts": [
+    "premium-data"
+  ]
+}
+EOF
+)
+```
+
+Use the app's key in an `apikey` query parameter
+in your request:
+
+```sh
+APIKEY="$(curl "https://apigee.googleapis.com/v1/organizations/$APIGEE_X_ORG/developers/testdatauser@example.com/apps/premium-data-app" \
+-H "Authorization: Bearer $APIGEE_TOKEN" | jq -r '.credentials[0].consumerKey')"
+curl "https://$APIGEE_X_HOSTNAME/london/bikes/v1?apikey=$APIKEY&try=[1-10]" -I
```

Comment:
Can we move `?apikey` to a header?

Reply:
This change will require an update to `VA-verify-apikey` and `AM-configure-defaults` to remove a header instead of q param